### PR TITLE
[enh] Enable app deployment via Git

### PR DESCRIPTION
**Pull request by Kloadut**
## Idea

Being able to `git push` to the proxy and deploy an app this way.

```
$ git remote add cozy https://mycozy.cozycloud.cc/repo/myapp
$ git push cozy master
Username for 'https://mycozy.cozycloud.cc': 
Password for 'https://my@email.org@mycozy.cozycloud.cc': 
remote: Counting objects: 75, done.
remote: Compressing objects: 100% (53/53), done.
remote: Total 62 (delta 27), reused 44 (delta 9)
remote: 
remote: info - Deploying 'myapp'
remote: info - Application 'myapp' has been deployed
remote:
Unpacking objects: 100% (62/62), done.
To https://mycozy.cozycloud.cc/repo/myapp
* [new branch]      master     -> cozy/master
```

## Details for cozy-proxy

I added the ability for the proxy to handle Git repositories via https://github.com/substack/pushover on the `/repo/*` routes.
I had to make a new authentication helper in order enable basic HTTP authentication with the Cozy owner's email as username, and its password as password. This way, Git ask for credentials natively at push, pull, and all remote operations.

To merge along with:
* https://github.com/cozy/cozy-home/pull/343
* https://github.com/cozy/cozy-monitor/pull/75
* https://github.com/cozy/cozy-controller/pull/58

**It lacks of test for now, please keep this PR open**

### CODE REVIEW

File: server/controllers/git.coffee [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    unless appName.match /^[a-zA-Z0-9-]{2,30}$/
+        res.statusCode = 400
+        return res.end "Bad request"
+
+    appRepo = "#{appsDir}/#{appName}"
+
+    req.url = req.url.substring "/repo".length
+    req.url = req.url.replace /\.git/, ""
+
+    if req.method is "POST"
```
Comment (frankrousseau) [inline server/controllers/git.coffee]: It's not the way to do with Express.

File: build/client/public/styles/app.css [hunk 2]
```diff
@@ -77,21 +77,21 @@ p {
   display: none;
 }
 .alert-error {
-  font-family: 'Maven Pro', sans-serif;
-  font-size: 20px;
+  font-family: 'Source Sans Pro', sans-serif;
```
Comment (frankrousseau) [inline build/client/public/styles/app.css]: When you change things that are not related to the PR, you should mention it somewhere.

Comment (Kloadut) [inline build/client/public/styles/app.css]: Oh, I didn't see that, sorry

File: server/controllers/git.coffee (hunk 1 above)
Comment (Kloadut) [inline server/controllers/git.coffee]: This line is actually not required, I will remove it

**Comment by clochix** [comment]
Hello @Kloadut,
You know what ? I'd like to know the status of this PR ! Thanks for updating it.

**Comment by Kloadut** [comment]
Need some additions
